### PR TITLE
Support remote files API

### PR DESCRIPTION
**Pull request by dleviminzi**
I would like to revive @ledmonster's effort to add remote file support.

### CODE REVIEW

File: remotefiles.go [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+
+	if err != nil {
+		return nil, err
+	}
+
+	return &response.RemoteFile, response.Err()
+}
+
+// ListRemoteFiles retrieves all remote files according to the parameters given. Uses cursor based pagination.
+func (api *Client) ListRemoteFiles(params ListRemoteFilesParameters) ([]RemoteFile, *ListRemoteFilesParameters, error) {
```
Comment (kanata2) [inline remotefiles.go]: I think `ListRemoteFilesParameters` should be used only as an argument to `ListRemoteFiles`.

**Review by kanata2 — COMMENTED** [review]
Thanks.

Please see this comment and update GoDoc.
https://github.com/slack-go/slack/pull/859#issuecomment-800572380

**Comment by dleviminzi** [comment]
> Thanks.
> 
> Please see this comment and update GoDoc. [#859 (comment)](https://github.com/slack-go/slack/pull/859#issuecomment-800572380)

I made the change to the return parameters. I'm not sure what you mean when you ask me to update the GoDoc. Are some explanations not detailed enough or something else?

**Comment by kanata2** [comment]
@dleviminzi 

Please add official API documentation's link 🙏 

e.g.


https://github.com/slack-go/slack/blob/03f86be11aa50ac65d66f3917e250d3257389028/chat.go#L185-L189

**Review by kanata2 — APPROVED** [review]
LGTM. Thanks!

**Comment by dleviminzi** [comment]
> LGTM. Thanks!

Apologies, I forgot to update the example when I changed the return arguments.

**Review by kanata2 — APPROVED** [review]
👍

**Comment by dleviminzi** [comment]
> 👍

Hello, when will this be able to get merged?

**Comment by kanata2** [comment]
merged!

**Pull request merged**